tests/func/test_import.py

```diff
@@ -56,6 +56,28 @@ def test_import_git_file(erepo_dir, tmp_dir, dvc, scm, src_is_dvc):
     }
 
 
+def test_import_cached_file(erepo_dir, tmp_dir, dvc, scm):
+    src = "some_file"
+    dst = "some_file_imported"
+
+    erepo_dir.scm_gen({src: "hello"}, commit="add a regular file")
+    os.remove(erepo_dir.dvc.config.config_file)
+
+    with tmp_dir.chdir():
+        f = open(dst, "w")
+        f.write("hello")
+        f.close()
+        tmp_dir.dvc.add([fspath(tmp_dir / dst)])
```

Comment:
You are adding by full path, which will make dvc treat it as an external output. What you want is:
```
tmp_dir.dvc_gen({dst: "hello"})
```
looks like.

Reply:
Side note: using open + write + close is a bad style even without pathlib, where you should use `with open()` instead.